### PR TITLE
Add a docker file for ROS2

### PR DESCRIPTION
**Pull request by MaximilienNaveau**
[//]: # "Thanks for your contribution.  To make life of the reviewers easier,"
[//]: # "please give this pull request a meaningful title and provide the"
[//]: # "requested information below."

## Description

Add a docker file for the ROS2 installation on ubuntu18.04

[//]: # "Description of what you did.  If it is more complex, consider to add a"
[//]: # "'Summary' section."

## How I Tested

[//]: # "Explain how you tested your changes"

Not tested yet, @vincentberenz:
- how can I make sure this works?
- can you check the setup_ubuntu mechanics?
- how can I chain the docker build?

## I fulfilled the following requirements

[//]: # "Please make sure you followed these steps before requesting a review."
[//]: # "Check the boxes in the list below, when done."

- [x] All new code is formatted according to our style guide (for C++ run clang-format, for Python, run flake8 and fix all warnings).
- [ ] All new functions/classes are documented and existing documentation is updated according to changes.
- [x] No commented code from testing/debugging is kept (unless there is a good reason to keep it).

### CODE REVIEW

File: official/ubuntu_18_04/docker/ros2/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+#             ROS INSTALLATION
+####################################################
+
+# Setup Local.
+RUN locale-gen en_US en_US.UTF-8
+RUN update-locale LC_ALL=en_US.UTF-8 LANG=en_US.UTF-8
+RUN export LANG=en_US.UTF-8
+
+# Setup Sources.
+RUN apt update && sudo apt install curl gnupg2 lsb-release
```
Comment (jviereck) [inline official/ubuntu_18_04/docker/ros2/Dockerfile]: For the `apt install`, should there be a `apt install -y` flag `-y` to make sure it installs without asking the user for input? Same below.

**Review by jviereck — CHANGES_REQUESTED** [review]
Looks good without testing it.See note about the `-y` option.

**Comment by vincentberenz** [comment]
@MaximilienNaveau 

- how can I make sure this works?
    - the bamboo job should be updated
    - building the docker images locally
    - running the installation script in docker

- can you check the setup_ubuntu mechanics?
   - looks ok to me

- how can I chain the docker build?
   -  the laas Docker file should be based on ros2 rather than ros, at this line: https://github.com/machines-in-motion/ubuntu_installation_scripts/blob/mnaveau/ros2/official/ubuntu_18_04/docker/laas/Dockerfile#L3
   -  the ros2 file should be based on ros rather than core here (https://github.com/machines-in-motion/ubuntu_installation_scripts/blob/mnaveau/ros2/official/ubuntu_18_04/docker/ros2/Dockerfile#L3) (did I get it right you want both ros1 and ros2 installed ?)

I am happy to do all the above if you prefer

**Comment by MaximilienNaveau** [comment]
> I am happy to do all the above if you prefer

I would be happy if you do ^^

And yes we can have ro1 and 2 installed on the machines.
I would not merge this before it is actually tested and that things run with ROS2 on my machines first...

File: official/ubuntu_18_04/docker/ros2/Dockerfile (hunk 1 above)
Comment (MaximilienNaveau) [reply official/ubuntu_18_04/docker/ros2/Dockerfile]: Fair point I will change that

**Comment by vincentberenz** [comment]
@MaximilienNaveau @jviereck 

Just pushed on this branch. Ros and Ros2 are now installed from the same dockerfile. I tested and all the dockerfile built ok on my machine. I also run setup_ubuntu on my machine, and it went fine.

**Review by jviereck — APPROVED** [review]
This looks good. However, the file `official/ubuntu_18_04/docker/ros2/Dockerfile` should be removed from the PR again now that ros1 and ros2 are installed from the same docker file?

**Comment by vincentberenz** [comment]
> This looks good. However, the file `official/ubuntu_18_04/docker/ros2/Dockerfile` should be removed from the PR again now that ros1 and ros2 are installed from the same docker file?

Indeed ! Just deleted the folder

**Comment by MaximilienNaveau** [comment]
Great I merge this!!!

**Pull request merged**